diag_manager/fms_diag_reduction_methods.F90

```diff
@@ -0,0 +1,695 @@
+                                                              !! in the I, J, and K dimensions; also holds
+                                                              !! halo sizes in the I, and J dimensions
+    logical, intent(in) :: reduced_k_range !< Flag indicating if the range in the K dimension is present
+
+    integer :: is, js, ks !< Starting indices in the I, J, and K dimensions
+    integer :: ie, je, ke !< Ending indices in the I, J, and K dimensions
+    integer :: hi, hj !< Halo sizes in the I, and J dimensions
+    integer :: f1, f2 !< Updated starting and ending indices in the I dimension
+    integer :: f3, f4 !< Updated starting and ending indices in the J dimension
+
+    !> Unpack bounds (/is, js, ks, ie, je, ke, hi, f1, f2, hj, f3, f4/)
+    is = recon_bounds%get_bounds3D()%get_imin()
+    js = recon_bounds%get_bounds3D()%get_jmin()
+    ks = recon_bounds%get_bounds3D()%get_kmin()
+    ie = recon_bounds%get_bounds3D()%get_imax()
+    je = recon_bounds%get_bounds3D()%get_jmax()
+    ke = recon_bounds%get_bounds3D()%get_kmax()
+    hi = recon_bounds%get_hi()
+    f1 = recon_bounds%get_fis()
+    f2 = recon_bounds%get_fie()
+    hj = recon_bounds%get_hj()
+    f3 = recon_bounds%get_fjs()
+    f4 = recon_bounds%get_fje()
+
+    ! Select proper type and update the buffer
+    select type (field_data)
+    type is (real(kind=r4_kind))
+      select type (buffer)
+      type is (real(kind=r4_kind))
+        if (flag .eq. 0) then
+        !> Update the buffer with the current minimum
+          if (reduced_k_range) then
+            ! recon_bounds must have ks = ksr and ke = ker
+            WHERE (mask(f1:f2,f3:f4,ks:ke,:) .AND. field_data(f1:f2,f3:f4,ks:ke,:) <&
+              buffer(is-hi:ie-hi,js-hj:je-hj,:,:,sample)) &
+              buffer(is-hi:ie-hi,js-hj:je-hj,:,:,sample) = field_data(f1:f2,f3:f4,ks:ke,:)
+          else
+            WHERE (mask(f1:f2,f3:f4,ks:ke,:) .AND. field_data(f1:f2,f3:f4,ks:ke,:) <&
+              buffer(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample)) &
+              buffer(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample) = field_data(f1:f2,f3:f4,ks:ke,:)
+          end if
+        else
+        !> Update the buffer with the current maximum
+          if (reduced_k_range) then
+            ! recon_bounds must have ks = ksr and ke = ker
+            WHERE (mask(f1:f2,f3:f4,ks:ke,:) .AND. field_data(f1:f2,f3:f4,ks:ke,:) <&
+              buffer(is-hi:ie-hi,js-hj:je-hj,:,:,sample)) &
+              buffer(is-hi:ie-hi,js-hj:je-hj,:,:,sample) = field_data(f1:f2,f3:f4,ks:ke,:)
+          else
+            WHERE (mask(f1:f2,f3:f4,ks:ke,:) .AND. field_data(f1:f2,f3:f4,ks:ke,:)>&
+              buffer(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample)) &
+              buffer(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample) = field_data(f1:f2,f3:f4,ks:ke,:)
+          end if
+        end if
+      class default
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_array_extremum type mismatch")
```

Comment:
What is mismatched?  Please provide more detail in these error messages.

Reply:
updated.